src/Compilers/CSharp/Portable/Symbols/MethodGroupConversionCaches/MethodGroupConversionCacheFrame.cs

```diff
@@ -0,0 +1,240 @@
+        public override ImmutableArray<TypeParameterSymbol> TypeParameters => BuiltTypeParameters;
+
+        private MethodGroupConversionCacheFrame(
+                MethodGroupConversionCacheFrameManager manager,
+                MethodSymbol targetMethod,
+                int arity)
+        {
+            Manager = manager;
+            SortKey = SymbolDisplay.ToDisplayString(targetMethod, SymbolDisplayFormat.FullyQualifiedFormat);
+
+            TypeParametersCount = arity;
+
+            var typeParamsBuilder = ArrayBuilder<TypeParameterSymbol>.GetInstance();
+            for ( int i = 0; i < arity; i++ )
+            {
+                typeParamsBuilder.Add(new MethodGroupConversionCacheFrameTypeParameter(this, i));
+            }
+            BuiltTypeParameters = typeParamsBuilder.ToImmutableAndFree();
+
+            var fieldType = TypeParameters[TypeParameters.Length - 1];
+            var fieldName = GeneratedNames.MakeMethodGroupConversionCacheDelegateFieldName(targetMethod.Name);
+            DelegateField = new SynthesizedFieldSymbol(this, fieldType, fieldName, isPublic: true, isStatic: true);
+        }
+
+        /// <summary>
+        /// Assign the name and index to be serialized in the module.
+        /// </summary>
+        /// <remarks>This method is only intended to be called from <see cref="MethodGroupConversionCacheFrameManager"/></remarks>
+        internal void AssignNameAndIndex(string name, int index)
+        {
+            _name = name;
+            _index = index;
+        }
+
+        /// <remarks>This method is only intended to be called from <see cref="MethodGroupConversionCacheFrameManager"/></remarks>
+        internal static MethodGroupConversionCacheFrame Create(MethodGroupConversionCacheFrameManager manager, MethodSymbol targetMethod)
+        {
+            Debug.Assert(manager != null);
+            Debug.Assert(targetMethod.IsDefinition);
+
+            return new MethodGroupConversionCacheFrame(manager, targetMethod, CalculateNeededArity(targetMethod));
+        }
+
+        private static int CalculateNeededArity(MethodSymbol targetMethod)
+        {
+            var arity = 1; // the type of the delegate
+
+            arity += targetMethod.Arity;
+
+            for (var containingType = targetMethod.ContainingType; containingType!= null; containingType = containingType.ContainingType)
+            {
+                arity += containingType.Arity;
+            }
+
+            return arity;
+        }
+
+        public static ImmutableArray<TypeSymbol> GetTypeArgumentsFromConversion(TypeSymbol conversionType, MethodSymbol targetMethod)
+        {
+            var typeArgumentsBuilder = ArrayBuilder<TypeSymbol>.GetInstance();
+
+            typeArgumentsBuilder.Add(conversionType);
+
+            if (targetMethod.Arity > 0)
+            {
+                var args = ((ConstructedMethodSymbol)targetMethod).TypeArguments;
+                AddTypeArgumentsReversed(typeArgumentsBuilder, ref args);
```

Comment:
No, not necessary to reverse, just reflected from the idea how I thought about the cache frame can be constructed to hold the cached instances. As long as the arity is correct, position is not significant.